src/tests/system/tests/test_netgroups.py

```diff
@@ -108,3 +108,142 @@ def test_netgroups__add_remove_netgroup_member(client: Client, provider: Generic
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_netgroup__user_attribute_membernisnetgroup_uses_group_dn(
+    client: Client, provider: GenericProvider, operation: str
+):
+    """
+    :title: User's 'memberNisNetgroup' attribute values are the DN of the group.
+    :setup:
+        1. Create users, groups.
+        2. Create a new netgroup called QAUsers and add a member (ng1) to QAUsers
+        3. Create another netgroup named DEVUsers and add a member (ng2) to DEVUsers
+        4. Modify the DEVUsers netgroup to replace its members with the members of QAUsers.
+        5. Start sssd
+    :steps:
+        1. Retrieve all members of the DEVUsers netgroup.
+        2. Confirm that the member directly added to DEVUsers is present.
+        3. Confirm that the member from QAUsers is now part of DEVUsers.
+    :expectedresults:
+        1. All members should be retrieved
+        2. Members directly added to DEVUsers is present.
+        3. Members from QAUsers is now part of DEVUsers.
+    :customerscenario: False
+    """
+    for id in [1, 2]:
+        provider.user(f"ng{id}").add()
+
+    netgroup_qa = provider.netgroup("QAUsers").add()
+    netgroup_qa.add_member(host="testhost1", user="ng1", domain="ldap.test")
+
+    netgroup_dev = provider.netgroup("DEVUsers").add()
+    netgroup_dev.add_member(host="testhost2", user="ng2", domain="ldap.test")
+    if operation == "Replace":
+        netgroup_dev.add_member(ng=netgroup_qa.dn)
+    else:
+        netgroup_dev.add_member(ng="QAUsers")
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost2, ng2, ldap.test)" in member
+    assert "(testhost1, ng1, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_netgroup__lookup_nested_groups(client: Client, provider: GenericProvider):
+    """
+    :title: Nesting netgroups and verifying user memberships using LDAP with sssd.
+    :setup:
+        1. Create users, groups.
+        2. Create netgroup named netgroup and Add Member
+        3. Create another netgroup named nested_netgroup
+        4. Add Members to nested_netgroup
+        5. Add Circular Netgroup Nesting to nested_netgroup
+        6. Start sssd
+    :steps:
+        1. Retrieves all members of the "nested_netgroup" group using the getent netgroup tool.
+        2. Verify that users from another group is also part of "nested_netgroup".
+        3. Checks if a user who is not in any netgroup is part of "nested_netgroup".
+        4. After the SSSD restart, it retrieves the members of "nested_netgroup" again to ensure they still intact.
+    :expectedresults:
+        1. All members of the "nested_netgroup" group be there
```

Comment:
"Netgroup is found, and both netgroups and users are members"

We can drop all the other steps.